content/3_Silver/2P.mdx

```diff
@@ -31,6 +31,143 @@ frequency: 2
 	</Resource>
 </Resources>
 
+## Two Pointers
+
+The two pointers method iterates two pointers across an array, to track the start and end
+of an interval. It can also be used to track two values in order to satisfy a specific condition. 
+
+<FocusProblem problem="book" />
+
+## Solution - Books
+
+We want to find the longest continuous segment of books that can be read within $t$ minutes. 
+
+To accomplish this, we can define $\texttt{left}$ and $\texttt{right}$ as the start and end of the segment respectively.
+
+Since being able to read segment $\texttt{left} \dots \texttt{right}$ implies we can read segment $\texttt{left} \dots \texttt{right}-1$, we can use two-pointers (or, alternatively, [binary search](/silver/binary-search)).
+
+For every $\texttt{left}$ index, we need to increase our $\texttt{right}$ pointer until the total time for that segment exceeds $t$.
+
+We will keep track of the max segment size with $\texttt{ans}$, and take the maximum $\texttt{right} - \texttt{left}$ value.
+
+We can then remove the $\texttt{left}$-indexed book from $\texttt{cur}$, the total time of current segment, and increment $\texttt{left}$ by 1.
```

Comment:
Hard to follow. Instead focus on the continuous movement of the left pointer and at each moment the calculation of max books read.

> We can then remove the $\texttt{left}$-indexed book from $\texttt{cur}$, the total time of current segment, and increment $\texttt{left}$ by 1.